### PR TITLE
Fixes #27331 - Fix pagination on policy dashboard

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ares** [comment]
rubocop needs some updates, feel free to add additional commit to this PR, I'll merge at once

**Comment by ares** [comment]
There are still some rubocop failures :-(

```
Inspecting 109 files
......................C.C..CC............................................C...................................

Offenses:

/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:8:15: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    return if @arf_report.logs.empty?
              ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:9:30: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    form_tag arf_report_path(@arf_report), :id => 'level_filter', :method => :get, :class => "form form-horizontal" do
                             ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:17:8: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    if @arf_report && @arf_report.host
       ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:17:23: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    if @arf_report && @arf_report.host
                      ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:23:35: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
                    { :caption => @arf_report.host.name,
                                  ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:24:48: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
                      :url => (arf_report_path(@arf_report) if authorized_for(hash_for_arf_report_path(@arf_report))) }
                                               ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/arf_reports_helper.rb:24:104: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
                      :url => (arf_report_path(@arf_report) if authorized_for(hash_for_arf_report_path(@arf_report))) }
                                                                                                       ^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/compliance_hosts_helper.rb:16:5: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    @host.arf_reports.of_policy(policy_id).each do |report|
    ^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:3:12: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    return @scap_content.scap_content_profiles if @scap_content.present?
           ^^^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:3:51: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    return @scap_content.scap_content_profiles if @scap_content.present?
                                                  ^^^^^^^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:4:12: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    return @policy.scap_content.scap_content_profiles if @policy.scap_content.present?
           ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:4:58: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    return @policy.scap_content.scap_content_profiles if @policy.scap_content.present?
                                                         ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:71:31: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
               { :selected => @policy.scap_content_profile_id },
                              ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:77:31: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
               { :selected => @policy.scap_content_profile_id,
                              ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:122:5: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    @policy.current_step == step ? 'show-pane' : 'hide-pane'
    ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:128:99: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
      link_to(previous.html_safe, '#', :class => 'btn btn-default', :onclick => "previous_step('#{@policy.previous_step}')")
                                                                                                  ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:141:8: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
    if @policy
       ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:147:35: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
                    { :caption => @policy.name,
                                  ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:148:49: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
                      :url => (edit_policy_path(@policy) if authorized_for(hash_for_edit_policy_path(@policy))) }
                                                ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policies_helper.rb:148:102: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
                      :url => (edit_policy_path(@policy) if authorized_for(hash_for_edit_policy_path(@policy))) }
                                                                                                     ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policy_dashboard_helper.rb:25:24: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
      content_tag(:h4, @report[label])
                       ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policy_dashboard_helper.rb:52:45: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
        _("Hosts no longer assigned: %s") % @report[:unassigned_hosts],
                                            ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/app/helpers/policy_dashboard_helper.rb:53:58: C: Rails/HelperInstanceVariable: Do not use instance variables in helpers.
        hosts_path(:search => "removed_from_policy = \"#{@policy.name}\"")
                                                         ^^^^^^^
/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.6/plugin/lib/foreman_openscap/helper.rb:14:8: C: Performance/InefficientHashSearch: Use #key? instead of #keys.include?.
    if Facets.registered_facets.keys.include?(:subscription_facet)RuboCop failed!

       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: app/views/policy_dashboard/_policy_reports.html.erb [hunk 1]
```diff
@@ -11,7 +11,7 @@
       <th><%= _("Other") %></th>
       <th><%= _('Actions') %></th>
     </tr>
-    <% @policy.arf_reports.latest.each do |arf_report| %>
+    <% @latest_reports.each do |arf_report| %>
```
Comment (ares) [inline app/views/policy_dashboard/_policy_reports.html.erb]: doesn't this reintroduce the behavior changed by https://github.com/theforeman/foreman_openscap/pull/377 ?

Comment (xprazak2) [reply app/views/policy_dashboard/_policy_reports.html.erb]: It does revert this particular line, but I cannot remember why it was changed in the first place. It should return the same results though as it calls `latest` scope for the reports belonging to a policy, I added a test. The only difference is `@policy.arf_reports.latest` gives me `undefined method `total_entries' for #<ForemanOpenscap::ArfReport::ActiveRecord_AssociationRelation:0x007efd8d475b58>` so I cannot paginate on that.

**Comment by ares** [comment]
Thanks make sense, fixes the issue. Merging!

**Pull request merged**